### PR TITLE
Add macOS graph switch shortcuts

### PR DESCRIPTION
**Pull request by maccman**
## Problem
Switching between recent graphs was mouse-driven: users had to open the sidebar graph dropdown and pick a row. On macOS that left a gap for keyboard-native graph switching, even though the app already centralizes navigation shortcuts through the command registry.

## Before -> After
| Before | After |
| --- | --- |
| Recent graphs could be selected from the sidebar dropdown only. | `Command+1` through `Command+9` switch to the matching recent graph row. |
| The graph dropdown did not teach the number shortcuts. | Non-current graph rows now show their matching shortcut hint beside the graph name. |
| The shortcut matcher only resolved `Mod-` bindings. | It now resolves `Meta-` bindings first, so graph switching is Command-key specific and `Ctrl+number` stays inert. |
| Digit shortcuts depended only on the produced character. | Number-row graph shortcuts also fall back to the physical `DigitN` code, so layouts that produce symbols or require Shift for digits still switch graphs. |

## Changes
1. Added `graph.switch1` through `graph.switch9` app commands bound to `Meta-1` through `Meta-9`.
2. Extended `CommandContext` with `switchGraph(index)` and wired `useAppShortcuts` to use the existing serialized `openRecent` flow against the current recents list.
3. Refactored shortcut dispatch into named lookup helpers for produced keys, physical digit fallback, and modifier prefixes, preserving existing punctuation shortcuts like history brackets.
4. Added graph-switch shortcut hints to the sidebar graph switcher rows, deriving them from the command registry.
5. Updated command-context and sidebar test harnesses and added coverage for command bindings, recent-graph index selection, current-graph no-op behavior, missing-slot no-op behavior, `Ctrl+number` not switching graphs, symbol-producing keyboard layouts, Shift-required digit layouts, and graph-menu keycap rendering.

## Tests
- `pnpm --filter @reflect/desktop test --run src/lib/commands/app-commands.test.ts src/routing/app-shortcuts.test.tsx`
- `pnpm --filter @reflect/desktop test --run src/lib/commands/registry.test.ts src/lib/attach-files.test.ts src/components/command-palette/command-palette.test.tsx src/components/sidebar/sidebar.test.tsx`
- `pnpm --filter @reflect/desktop test --run src/routing/app-shortcuts.test.tsx`
- `pnpm --filter @reflect/desktop test --run src/routing/app-shortcuts.test.tsx src/components/sidebar/sidebar.test.tsx`

## Verification
- `pnpm check` passed. It reported an existing max-lines warning in `packages/core/src/indexing/indexer.ts`, outside this diff.

## Risk / Rollout
No migrations or data changes. The switch uses the existing `openRecent` graph-open path, so it inherits the current serialization and graph-session guards. The main behavioral risk is shortcut matching; the PR pins the modifier and keyboard-layout cases with focused tests.

---

> [!NOTE]
> **Medium Risk**
> Touches global shortcut resolution for all app commands; wrong matching could fire unintended commands, though tests cover Meta-only gating and keyboard-layout edge cases.
> 
> **Overview**
> Adds **keyboard switching between recent graphs** on macOS: **⌘1–⌘9** map to the first nine rows in the recents list and call the existing **`openRecent`** flow (no-op when that graph is already open or the slot is empty).
> 
> Registers **`graph.switch1`–`graph.switch9`** in the command registry with **`Meta-N`** bindings and extends **`CommandContext`** with **`switchGraph(index)`**. The sidebar graph menu shows matching shortcut hints on non-current recent rows via **`keybindingFor`**.
> 
> **Shortcut dispatch** now tries **`Meta-`**, **`Ctrl-`**, and **`Mod-`** prefixes and adds a **physical `DigitN` fallback** so number-row shortcuts still work on symbol-producing or Shift-for-digit layouts, while **`Ctrl+number`** does not switch graphs.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit ae68b971ba8e9feb64ac74617854e960449212aa. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

## Summary by CodeRabbit

* **New Features**
  * Added “switch to recent graph” commands bound to ⌘+1 through ⌘+9.
  * Updated the graph switch dropdown to display the shortcut keys for each recent graph.
* **Bug Fixes**
  * Prevented graph switching from navigating when the target recent graph is already open.
  * Improved number-shortcut handling across keyboard layouts/modifier states, with physical digit awareness and correct ⌘ gating.
* **Tests**
  * Expanded/updated coverage for graph-switch command execution, keybinding mappings, and recent-graph shortcut routing.

### CODE REVIEW

File: apps/desktop/src/routing/app-shortcuts.ts [hunk 1]
```diff
@@ -45,10 +45,6 @@ const CODE_TO_BINDING_KEY: Record<string, string> = {
   BracketRight: ']',
 }
 
-function isModKey(event: KeyboardEvent): boolean {
-  return event.metaKey || event.ctrlKey
-}
-
```
[the remark this answers is not in the document]
Comment (maccman) [reply apps/desktop/src/routing/app-shortcuts.ts:51]: Fixed in 85ec34f4 by falling back to the physical `DigitN` code after produced-character matching, with coverage for a symbol-producing `Digit2` layout.

File: apps/desktop/src/routing/app-shortcuts.ts [hunk 2]
```diff
@@ -68,15 +64,39 @@ function bindingKeyFor(event: KeyboardEvent): string {
...
+      `${alt}Mod-${shift}${key}`,
+    ].filter((candidate): candidate is string => candidate !== null)
+    for (const candidate of candidates) {
+      const id = BINDING_TO_ID.get(candidate)
+      if (id !== undefined) {
+        return id
+      }
+    }
+  }
+  return null
```
[the remark this answers is not in the document]
Comment (maccman) [reply apps/desktop/src/routing/app-shortcuts.ts:134]: Fixed in 5db5e65d by giving the physical digit fallback its own unshifted lookup path, while leaving the produced-key path unchanged. Added tests for Shift-required digits and for shifted symbols staying inert.
